components/searchable_user_list/searchable_user_list.tsx

```diff
@@ -2,46 +2,47 @@
 // See LICENSE.txt for license information.
 /* eslint-disable react/no-string-refs */
 
-import $ from 'jquery';
-import PropTypes from 'prop-types';
-import React from 'react';
-import ReactDOM from 'react-dom';
+import React, {createRef, ReactNode} from 'react';
+
 import {FormattedMessage, injectIntl} from 'react-intl';
 
 import QuickInput from 'components/quick_input';
 import UserList from 'components/user_list.jsx';
 import LocalizedInput from 'components/localized_input/localized_input';
 
 import {t} from 'utils/i18n';
+import {IntlProps} from 'components/search_results/types';
 
 const NEXT_BUTTON_TIMEOUT = 500;
 
-class SearchableUserList extends React.PureComponent {
-    static propTypes = {
-        users: PropTypes.arrayOf(PropTypes.object),
-        usersPerPage: PropTypes.number,
-        total: PropTypes.number,
-        extraInfo: PropTypes.object,
-        nextPage: PropTypes.func.isRequired,
-        previousPage: PropTypes.func.isRequired,
-        search: PropTypes.func.isRequired,
-        actions: PropTypes.arrayOf(PropTypes.func),
-        actionProps: PropTypes.object,
-        actionUserProps: PropTypes.object,
-        focusOnMount: PropTypes.bool,
-        renderCount: PropTypes.func,
-        filter: PropTypes.string,
-        renderFilterRow: PropTypes.func,
-        page: PropTypes.number.isRequired,
-        term: PropTypes.string.isRequired,
-        onTermChange: PropTypes.func.isRequired,
-        intl: PropTypes.any,
-        isDisabled: PropTypes.bool,
-
-        // the type of user list row to render
-        rowComponentType: PropTypes.func,
-    };
-
+ type Props = {
+     users: Array<Record<string, unknown>>;
+     usersPerPage: number;
+     total: number;
+     extraInfo?: Record<string, unknown>;
+     nextPage: (page?: number) => Promise<void> | void;
+     previousPage: (page?: number) => Promise<void> | void;
+     search: (term: string) => void;
+     actions?: ReactNode[];
+     actionProps?: Record<string, unknown>;
+     actionUserProps?: Record<string, unknown>;
+     focusOnMount?: boolean;
+     renderCount?: (count: number, total: number, startCount: number, endCount: number, isSearch: boolean) => ReactNode;
+     filter?: string;
+     renderFilterRow?: (func: {(e: React.FormEvent<HTMLInputElement>): void}) => void;
+     page: number;
+     term: string | number;
+     onTermChange: (term: string) => void;
+     intl: (msg: Record<string, string>) => string;
+     isDisabled?: boolean;
+
+     // the type of user list row to render
+     rowComponentType?: ReactNode;
```

Comment:
This should also be `ElementType`

Reply:
I think it's actually `ComponentType`, not fully sure though.